### PR TITLE
Mini drawer

### PR DESCRIPTION
**Pull request by ingarabr**
I ported the mini-drawer and bumped into some interesting issues. I'm a but clueless on how to resolve them so some help had been much appreciated. 

Remaining work:
- [x] Figure out why the open and close animation isn't working. Not sure why but it apairs to be related to some child nodes being recreated.
- [x] The position of the icons changes position when the drawer is open. It looks like it's related to padding values that are applied and might be related to the missing styling (see inline todos).
- [x] The mini-drawer demo includes styling that isn't "typesafe". Figure out how we can apply it.
  `.set("class", ...)`
- [x]  Get hold of the theme to set the right icon based on the placement of the drawer (see inline todo).

### CODE REVIEW

File: src/main/scala/demo/MiniDrawer.scala [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        boxSizing = "border-box"
+        `MuiDrawer-paper` =
+          if (open) openedMixin(theme)
+          else closedMixin(theme)
+      }.combineWith(
+        if (open) openedMixin(theme)
+        else closedMixin(theme)
+      )
+    )
+    .build()
```
Comment (oyvindberg) [inline src/main/scala/demo/MiniDrawer.scala:114]: save the result of `build` as a `val` and use that as a react component

File: src/main/scala/demo/MiniDrawer.scala [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+              if (open) display = "none"
+            })(muiIcon.Menu()),
+          mui.Typography.variant("h6").noWrap(true).component("div")("Mini variant drawer")
+        )
+      ),
+      Drawer(open)(
+        DrawerHeader(
+          mui.IconButton.normal
+            .onClick(handleDrawerClose)(
+              muiIcon.ChevronLeft() // todo: how to get hold of the theme? {theme.direction === 'rtl' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
```
Comment (oyvindberg) [inline src/main/scala/demo/MiniDrawer.scala]: https://mui.com/material-ui/customization/theming/#accessing-the-theme-in-a-component

File: src/main/scala/demo/MiniDrawer.scala [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+      "width",
+      new Partial {
+        easing = theme.transitions_BaseTheme.easing.sharp
+        duration = theme.transitions_BaseTheme.duration.leavingScreen
+      }
+    )
+    overflowX = "hidden"
+    width = s"calc(${theme.spacing(7)} + 1px)"
+
+    // todo: figure out how to apply this
```
Comment (oyvindberg) [inline src/main/scala/demo/MiniDrawer.scala]: you can call `.set(name, value)` on `CSSObject`, so something like 
```scala
.set(theme.breakpoints.up("sm"), new js.Object {val width =  s"calc(${theme.spacing(8)} + 1px")})
```

**Comment by oyvindberg** [comment]
Did any of those changes help?

File: src/main/scala/demo/MiniDrawer.scala (hunk 1 above)
Comment (ingarabr) [reply src/main/scala/demo/MiniDrawer.scala:114]: Got into an issue where we need to set some values based on some props. We found a workaround where we did an unsafe set and cast the value needed within the `fn` method.

**Comment by oyvindberg** [comment]
nice, thanks! ready to merge right away?

**Comment by ingarabr** [comment]
Jepp, it's ready to be merged.

**Pull request merged**